modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/MultiSelectTag.java

```diff
@@ -0,0 +1,232 @@
+
+		return super.doStartTag();
+	}
+
+	public String getClearAllTitle() {
+		return _clearAllTitle;
+	}
+
+	public String getCloseButtonAriaLabel() {
+		return _closeButtonAriaLabel;
+	}
+
+	public boolean getDisabledClearAll() {
+		return _disabledClearAll;
+	}
+
+	public Object getFilter() {
+		return _filter;
+	}
+
+	public String getInputName() {
+		return _inputName;
+	}
+
+	public String getInputValue() {
+		return _inputValue;
+	}
+
+	public boolean getIsValid() {
+		return _isValid;
+	}
+
+	public ArrayList<Object> getItems() {
+		return _items;
+	}
+
+	public Object getLocator() {
+		return _locator;
+	}
+
+	public ArrayList<Object> getSourceItems() {
+		return _sourceItems;
+	}
+
+	public boolean isDisabled() {
+		return _disabled;
+	}
+
+	public void setClearAllTitle(String clearAllTitle) {
+		_clearAllTitle = clearAllTitle;
+	}
+
+	public void setCloseButtonAriaLabel(String closeButtonAriaLabel) {
+		_closeButtonAriaLabel = closeButtonAriaLabel;
+	}
+
+	public void setDisabled(boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setDisabledClearAll(boolean disabledClearAll) {
+		_disabledClearAll = disabledClearAll;
+	}
+
+	public void setFilter(Object filter) {
+		_filter = filter;
+	}
+
+	public void setInputName(String inputName) {
+		_inputName = inputName;
+	}
+
+	public void setInputValue(String inputValue) {
+		_inputValue = inputValue;
+	}
+
+	public void setIsValid(boolean isValid) {
+		_isValid = isValid;
+	}
+
+	public void setItems(ArrayList<Object> items) {
+		_items = items;
+	}
+
+	public void setLocator(Object locator) {
+		_locator = locator;
+	}
+
+	public void setSourceItems(ArrayList<Object> sourceItems) {
+		_sourceItems = sourceItems;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_clearAllTitle = null;
+		_closeButtonAriaLabel = null;
+		_disabled = false;
+		_disabledClearAll = false;
+		_filter = null;
+		_inputName = null;
+		_inputValue = null;
+		_isValid = false;
+		_items = null;
+		_locator = null;
+		_sourceItems = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/MultiSelect";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("clearAllTitle", _clearAllTitle);
+		props.put("closeButtonAriaLabel", _closeButtonAriaLabel);
+		props.put("disabledClearAll", _disabledClearAll);
+		props.put("disabled", _disabled);
+		props.put("filter", _filter);
+		props.put("inputName", _inputName);
+		props.put("inputValue", _inputValue);
+		props.put("isValid", _isValid);
+		props.put("items", _items);
+		props.put("locator", _locator);
+		props.put("sourceItems", _sourceItems);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("input-group-item");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
```

Comment:
Missing all the items rendering here.

Reply:
Okay, I initially assumed it's gonna get completely handled in Javascript, but I guess for those items that get pulled from the server/db.

Reply:
You can receive `selectedItems` in the tag, so it's more than likely that those items already selected will be passed to the tag by the server, so yep, you need to render them here :)